internal/host/host.go

```diff
@@ -309,15 +309,22 @@ func (m *Manager) UpdateInventory(ctx context.Context, h *models.Host, inventory
 }
 
 func (m *Manager) RefreshStatus(ctx context.Context, h *models.Host, db *gorm.DB) error {
+	log := logutil.FromContext(ctx, m.log)
 	if db == nil {
 		db = m.db
 	}
 	vc, err := newValidationContext(h, db)
 	if err != nil {
+		log.WithError(err).Errorf("Failed to create new validation context for host %s", *h.ID)
 		return err
 	}
 	conditions, validationsResults, err := m.rp.preprocess(vc)
 	if err != nil {
+		log.WithError(err).Errorf("Failed to preprocess validations for host %s", *h.ID)
+		return err
+	}
+	if err = m.reportValidationChangedMetrics(ctx, vc, h, validationsResults); err != nil {
```

Comment:
mmm will we fail if we could not report metric? What will it affect? Not sure this is a desired behavior

Reply:
@filanov

Reply:
We already had a pretty similar discussion on it https://github.com/openshift/assisted-service/pull/885#discussion_r559199985 @romfreiman

Reply:
From the code we don't fail on reporting, just on the parsing of the validations struct, i agree that metrics should not affect refresh flow.

Reply:
Don't forget that reporting metrics doesn't rely on any external service.  All it needs is that the DB doesn't fail and inventory can be unmarshalled.  There's a tradeoff between having unreliable metrics and disrupting the service unnecessarily.  In this case I think that if submitting metrics isn't working, the service isn't working either, so it's a moot point.  Probably better to fail on it just so it doesn't hide bugs.

Reply:
This is the same case, here as well we fail only on parsing the validations struct, we won't fail in other cases.

Reply:
@avishayt how won't it hide bugs? We can write a traceback and monitor thoses tracebacks. But failing user operation will not actually help, right? He'll delete the cluster and retry. And, we'll in the best case, will find about such failure couple of days later. So not sure how failure will help. If service is broken, it should fail on the state update as well.

Reply:
I mean that if the DB isn't working or validations can't be parsed, then nothing will work anyway.  I see it as very low risk.  But initially I felt the same way and made the same comment as you.

Reply:
The concern here is that if someone will add more logic to `reportValidationChangedMetrics` and something will fail we will break refresh status. beside that it's ok